tikv-client/src/main/java/com/pingcap/tikv/statistics/IndexStatistics.java

```diff
@@ -0,0 +1,112 @@
+package com.pingcap.tikv.statistics;
+
+import com.google.common.collect.BoundType;
+import com.google.common.collect.Range;
+import com.pingcap.tikv.codec.CodecDataOutput;
+import com.pingcap.tikv.key.Key;
+import com.pingcap.tikv.key.TypedKey;
+import com.pingcap.tikv.meta.TiIndexInfo;
+import com.pingcap.tikv.predicates.IndexRange;
+import com.pingcap.tikv.types.DataTypeFactory;
+import com.pingcap.tikv.types.MySQLType;
+
+import java.util.List;
+
+public class IndexStatistics {
+  private Histogram histogram;
+  private CMSketch cmSketch;
+  private TiIndexInfo indexInfo;
+
+  public IndexStatistics(Histogram histogram, CMSketch cmSketch, TiIndexInfo indexInfo) {
+    this.histogram = histogram;
+    this.cmSketch = cmSketch;
+    this.indexInfo = indexInfo;
+  }
+
+  public Histogram getHistogram() {
+    return histogram;
+  }
+
+  public void setHistogram(Histogram histogram) {
+    this.histogram = histogram;
+  }
+
+  public CMSketch getCmSketch() {
+    return cmSketch;
+  }
+
+  public void setCmSketch(CMSketch cmSketch) {
+    this.cmSketch = cmSketch;
+  }
+
+  public TiIndexInfo getIndexInfo() {
+    return indexInfo;
+  }
+
+  public void setIndexInfo(TiIndexInfo indexInfo) {
+    this.indexInfo = indexInfo;
+  }
+
+  public double getRowCount(List<IndexRange> indexRanges) {
+    double rowCount = 0.0;
+    for (IndexRange ir : indexRanges) {
+      double cnt = 0.0;
+      Key pointKey = ir.hasAccessKey() ? ir.getAccessKey() : Key.EMPTY;
+      Range<TypedKey> range = ir.getRange();
+      Key lPointKey;
+      Key uPointKey;
+
+      Key lKey;
+      Key uKey;
+      if (pointKey != Key.EMPTY) {
+        Key convertedKey = TypedKey.toTypedKey(pointKey.getBytes(), DataTypeFactory.of(MySQLType.TypeBlob));
+        Key convertedNext = TypedKey.toTypedKey(pointKey.next().getBytes(), DataTypeFactory.of(MySQLType.TypeBlob));
+        // TODO: Implement CMSketch point query
+//        if (cmSketch != null) {
+//          rowCount += cmSketch.queryBytes(convertedKey.getBytes());
```

Comment:
There's still some issue implementing CMSketch query, mainly because Java does not support `unsigned long`.